pkg/cache/cluster.go

```diff
@@ -31,6 +32,33 @@ const (
 	ClusterRetryTimeout        = 10 * time.Second
 )
 
+var (
+	// listSemaphore is used to limit the number of concurrent k8s list queries.
+	// Global limit is required to avoid memory spikes during cache initialization.
+	// The default limit of 50 is chosen based on experiments.
+	listSemaphore = semaphore.NewWeighted(50)
+)
+
+// SetMaxConcurrentList set maximum number of concurrent K8S list calls.
```

Comment:
I think it should be documented that if `val` is less or equal to `0`, limit is disabled.

Reply:
done.